lib/fs.js

```diff
@@ -16,19 +16,22 @@ const mkdir = (dir, mode, cb) => {
   });
 };
 
-const checkVersion = () => {
+let checkVersion = () => {
   const version = process.versions.node.split('.').map(el => parseInt(el));
-  return version[0] < 10 || (version[0] === 10 && version[1] <= 11) ? 0 : 1;
+  if (version[0] < 10 || (version[0] === 10 && version[1] <= 11))
+    return (dir, modeValue, cb) => mkdir(dir, modeValue, cb);
+  else
+    return (dir, modeValue, cb) =>
+      fs.mkdir(dir, { recursive: true, mode: modeValue }, cb);
 };
 
-const variant = checkVersion();
-
-const mkdirp = (dir, modeValue, cb) => {
-  variant
-    ? fs.mkdir(dir, { recursive: true, mode: modeValue }, cb)
-    : mkdir(dir, modeValue, cb);
+let mkdirp = (dir, modeValue, cb) => {
+  checkVersion(dir, modeValue, cb);
 };
 
+checkVersion = checkVersion();
```

Comment:
I'd prefer to avoid unnecessary code:
```js
let mkdirp;
if (version[0] < 10 || (version[0] === 10 && version[1] <= 11)) {
  mkdirp = mkdir;
} else {
  mkdirp = (dir, mode, cb) => fs.mkdir(dir, { recursive: true, mode }, cb);
}
```

Reply:
And, as @belochub mentioned in https://github.com/metarhia/common/pull/249#discussion_r256514751 there is no need for a separate `checkVersion` function.